### ISSUE TITLE
Argument types do not match

### ISSUE THREAD

**Issue by pantonis** [AutoMapper/AutoMapper#3505]
### Source/destination types
```
 public abstract class Entity<TKey> : IEntity<TKey>
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public virtual TKey Id { get; set; }
    }

  public class Tenant : Entity<long>, IAuditedEntity
    {
        [Required]
        [MaxLength(ValidationDefinitions.NAME_FIELD_LENGTH)]
        public string Name { get; set; }

        [Required]
        public bool IsActive { get; set; }


        [ForeignKey("TimeZoneId")]
        public Timezones.TimeZone TimeZone { get; set; }
        public int? TimeZoneId { get; set; }

        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }

        [ForeignKey("CountryId")]
        public Country Country { get; set; }
        public int? CountryId { get; set; }
    }

    public class TenantDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreationDate { get; set; }
        public string Country { get; set; }
    }


```

### Mapping configuration
```
public class TenantProfile : Profile
    {
        public TenantProfile()
        {
            CreateMap<TenantRegisterInputDto, Tenant>();

            CreateMap<Tenant, TenantDto>()
                    .ForMember(dest => dest.Country, opt => opt.MapFrom(src => src.Country.Name));
        }
    }

```

### Expected behavior

Return mapped list

### Actual behavior

Got error ```Argument types do not match```

### Steps to reproduce

```csharp
 var queryable = tenantsRepository.GetAll().Include(x => x.Country).AsNoTracking();
List<TenantDto> tenants = await (from tenant in queryable
                                                        select tenant).ProjectTo<TenantDto>(mapper.ConfigurationProvider).ToListAsync();

```

**Comment by lbargaoanu** [comment]
Remove the `Include` and try again. If it still fails, a repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by pantonis** [comment]
@lbargaoanu I removed it and still the error occured. I found the issue is the ```public DateTime CreationDate { get; set; }```. I dont understand why though. Using ```mapper.Map<List<TenantDto>>(tenants);``` works like a charm

**Comment by lbargaoanu** [comment]
Once you have a repro, I'll take a look.

**Comment by pantonis** [comment]
I managed to debug (automapper version 9) and find the issue. I cant reproduce it in a small project and the reason is that there are hundreds of mappings and the _typeMapPlanCache is populated from other types in a large project.
Here is what I found:
The problem relies with a DateTime property as I said above.
In ```ExpressionBuilder.cs``` in the following method ```List<MemberBinding> CreateMemberBindings(ExpressionRequest request, TypeMap typeMap, Expression instanceParameter, TypePairCount typePairCount, LetPropertyMaps letPropertyMaps)```
the call to this method ```var propertyTypeMap = _configurationProvider.ResolveTypeMap(result.Type, propertyMap.DestinationType);``` returns an object with values
and the call to this method ``var binder = _configurationProvider.Binders.FirstOrDefault(b => b.IsMatch(propertyMap, propertyTypeMap, result));``` returns an object of type ```MappedTypeExpressionBinder```. 
When ```binder.bind``` is called it gets in BindMappedTypeExpression and when it reached the ```if (propertyMap.TypeMap.Profile.AllowNullDestinationValues && !propertyMap.AllowNull && !(result.ResolutionExpression is ParameterExpression) && !result.ResolutionExpression.Type.IsCollectionType())``` it steps in and in the line inside if it breaks.


With a small project:

In ```ExpressionBuilder.cs``` in the following method ```List<MemberBinding> CreateMemberBindings(ExpressionRequest request, TypeMap typeMap, Expression instanceParameter, TypePairCount typePairCount, LetPropertyMaps letPropertyMaps)```
the call to this method ```var propertyTypeMap = _configurationProvider.ResolveTypeMap(result.Type, propertyMap.DestinationType);``` returns null
and the call to this method ``var binder = _configurationProvider.Binders.FirstOrDefault(b => b.IsMatch(propertyMap, propertyTypeMap, result));``` returns an object of type ```AssignableExpressionBinder```.  where it does not have these checks and the object is successfully mapped
